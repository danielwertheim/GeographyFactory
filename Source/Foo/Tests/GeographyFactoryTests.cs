using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Foo.Tests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class GeographyFactoryTests
    {
        [Test]
        public virtual void When_Coordinates_has_correct_order_and_is_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.CorrectOrderAndClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_correct_order_and_is_not_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.CorrectOrderNotClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_and_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongOrderAndClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_and_not_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongOrderNotClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_starting_south_to_north_and_many_to_west_and_back_and_is_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongSouthToNorthAndManyWestAndBackClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Castle()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.Castle());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Star_drawn_from_middle_and_clockwise()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.StartThatDoesNotRenderWithoutFix());

            Approvals.Verify(g);
        }
    }
}