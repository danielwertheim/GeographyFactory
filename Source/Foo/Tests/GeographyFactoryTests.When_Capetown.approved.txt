POLYGON ((18.623324871063232 -33.87930689800001, 18.518954753875732 -33.814295035787055, 18.469516277313232 -33.906665509534392, 18.410464763641357 -33.904385960456537, 18.365146160125732 -33.929457647348386, 18.366519451141357 -33.97046812793053, 18.443423748016357 -34.00804370312774, 18.446170330047607 -34.072907819484428, 18.466769695281982 -34.103615608063791, 18.638431072235107 -34.0694951555098, 18.693362712860107 -34.0694951555098, 18.731814861297607 -34.043326831827528, 18.741427898406982 -33.978440092814679, 18.700229167938232 -33.915783096255439, 18.679629802703857 -33.880447015354235, 18.623324871063232 -33.87930689800001))